### PR TITLE
added RMSprop optim

### PR DESCRIPTION
**Pull request by f0ti**
Sorry for commenting out SGD, had to do the tests

### CODE REVIEW

**Comment by geohot** [comment]
Haha does it get a better loss than SGD?

**Comment by f0ti** [comment]
> Haha does it get a better loss than SGD?

No, it doesn't beat SGD :(

File: README.md [hunk 1]
```diff
@@ -57,7 +57,8 @@ class TinyBobNet:
     return x.dot(self.l1).relu().dot(self.l2).logsoftmax()
 
 model = TinyBobNet()
-optim = optim.SGD([model.l1, model.l2], lr=0.001)
+optim = tinygrad_optim.SGD([model.l1, model.l2], lr=0.001) # or
+optim = tinygrad_optim.RMSprop([model.l1, model.l2], lr=0.001)
```
Comment (geohot) [inline README.md:61]: Oh, I didn't mean here, I meant in the list of optimizers it supports.

File: test/test_mnist.py [hunk 2]
```diff
@@ -36,56 +36,63 @@ def forward(self, x):
...
+    train(model, optim, steps)
+    evaluate(model)
+    
+    model = TinyBobNet()
+    optim = tinygrad_optim.SGD([model.l1, model.l2], lr=0.001)
+    steps = 1000
+    train(model, optim, steps)
+    evaluate(model)
+    
+    # RMSprop
```
Comment (geohot) [inline test/test_mnist.py:92]: Hmm, isn't this using the weights from SGD?

File: README.md [hunk 3]
```diff
@@ -41,11 +41,11 @@ print(y.grad)  # dz/dy
 
 It turns out, a decent autograd tensor library is 90% of what you need for neural networks. Add an optimizer (SGD and Adam implemented) from tinygrad.optim, write some boilerplate minibatching code, and you have all you need.
 
-### Neural network example (from test/mnist.py)
+### Neural network example (from test/test_mnist.py)
 
 ```python
 from tinygrad.tensor import Tensor
-import tinygrad.optim as optim
```
Comment (geohot) [inline README.md:48]: I like to import as optim, it matches torch.

File: README.md (hunk 1 above)
Comment (f0ti) [reply README.md:61]: Oh, okk...

File: test/test_mnist.py (hunk 2 above)
Comment (f0ti) [reply test/test_mnist.py:92]: Yeah right, initialized new model for new optimizer

File: README.md (hunk 3 above)
Comment (f0ti) [reply README.md:48]: Done

**Pull request merged**